### ISSUE TITLE
is our block description syntax valid yaml?

### ISSUE THREAD

**Issue by williballenthin** [fireeye/capa#312]
currently, we allow rule authors to add a description to and/or/etc. like so:

```yaml
and:
  - match: foo
  - match: bar
  description: this is awesome!
```

(concrete example [here](https://github.com/fireeye/capa-rules/blob/433b32c1b003efe78ee38d8e7c8b795fa1fe2c76/data-manipulation/encryption/rc4/encrypt-data-using-rc4-ksa.yml#L26))

this seemed like a natural way to extend our two-line syntax for feature descriptions, which look like:

```yaml
and:
  - string: foo
    description: bar
```

however, i'm not sure that the former is valid yaml. ruamel accepts it, but pyyaml does not. take a moment to look at it and try to figure out how it should be parsed? is the child of the `and` a list or dictionary???

**Comment by williballenthin** [comment]
one way to make this work while only slightly tweaking the syntax is to make `description` an element in the list, like:

```yaml
and:
  - match: foo
  - match: bar
  - description: this is awesome!
```

**Comment by williballenthin** [comment]
downside: fixing up this syntax is technically a breaking change, affecting the last minor version or so. the alternative is to suffer having rules that are maybe not valid yaml (something I think is fairly important). 

personally, I probably prefer to keep things as we intended and make the change (violating semver and assuming no one will notice).

**Comment by williballenthin** [comment]
upside: the description can now be placed at the top of a block, like:

```yaml
and:
  - description: everything is awesome!
  - match: foo
  - match: bar
```

I think all our rules with the conflicting syntax have the description as the last element, which I personally find hard to follow, all because this is all that ruamel accepted.

**Comment by williballenthin** [comment]
maybe for v1.3.0 we should comment out these description lines as they currently exist so we can fully discuss the changes here while minimizing the time the invalid syntax exists.

**Comment by williballenthin** [comment]
tagging @re-fox since I know they have used this syntax to make rules easier to read

**Comment by mr-tz** [comment]
Yeah, let's fix this. Commenting out is a good start!

**Comment by re-fox** [comment]
I agree!  That sounds like a good plan.

I remember at the time, modifying the rule several times before I found a working variation.  I think that's how I ended up on.

```yaml
  - api: foo
  description: bar
```

In my opinion it's better to have commented descriptions than none at all.  Those can always be uncommented and fixed up later.
